pkg/osvscanner/osvscanner.go

```diff
@@ -124,9 +128,23 @@ type gitIgnoreMatcher struct {
 
 func parseGitIgnores(dir string) (*gitIgnoreMatcher, error) {
 	// We need to parse .gitignore files from the root of the git repo to correctly identify ignored files
-	// Defaults to current directory if dir is not in a repo or some other error
-	// TODO: Won't parse ignores if dir is not in a git repo, and is not under the current directory (e.g ../path/to)
-	fs := osfs.New(".")
+	var fs billy.Filesystem
+
+	// Default to dir (or directory containing dir if it's a file) is not in a repo or some other error
+	f, err := os.Open(dir)
```

Comment:
I think you can combine the Open and Stat into `os.Stat`

Reply:
Oh, I didn't see that function. Changed.